### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by adespawn**
Adds integration tests to CI.

Tools required to run the integration tests (not automatically installed):
 - [scylla-ccm python script](https://github.com/scylladb/scylla-ccm) used by tests to manage database instances 
 - [simulacron.jar](https://github.com/datastax/simulacron/) used by some tests. Currently tests assume that this file is in ``SIMULACRON_PATH`` directory if such env variable is provided or ``~/ `` otherwise. 

Some of the tests were removed as they test features that are not planed. Other were disabled due to lack of consistency or just because they don't work. Currently there should be
 - 491 passing tests
 
 All integration testes are in one category, compared to being split ino short and long in DataStax driver. Instead of that, there are split depending on the state of the supported features in the driver.

Additionally I wasn't able to do other things related to integration tests. For this reason I created following issues: #45 #44.
Probably there should be also other issue addressing currently removed tests. 

Thanks to the @wprzytula  suggestion I changed from datastax ccm to scylla ccm, which helped to solve problem mentioned in #45. Thanks to that I also updated README with instructions on running the tests.

### CODE REVIEW

**Comment by wprzytula** [comment]
> * [ccm python script](https://github.com/riptano/ccm) used by tests to manage database instances

I advice using https://github.com/scylladb/scylla-ccm for best ScyllaDB compatibility.

**Comment by wprzytula** [comment]
> * [simulacron.jar](https://github.com/datastax/simulacron/) used by some tests. Currently tests assume that this file is in `~/ ` directory. (It's possible to change the directory to something else: maybe `~/.local/lib/`?

Maybe we could keep the simulacron in our repo? Or at least provide a script that automatically fetches it for use in tests, to some directory local to the driver's repo.

**Comment by wprzytula** [comment]
> Probably there should be also other issue addressing currently removed tests.

Definitely!
You can try the sub-issues mechanism for that. You create a task list, like this:
- [ ] task1
- [ ] task2

And then you can just click "convert to issue" next to each task. This way you quickly create an umbrella issue and its children.

File: package.json [hunk 1]
```diff
@@ -47,7 +47,9 @@
     "build": "napi build --platform --release",
     "test": "./node_modules/.bin/mocha test/unit -R spec -t 5000 --recursive",
     "unit": "./node_modules/.bin/mocha test/unit -R spec -t 5000 --recursive",
-    "examples": "node ./examples/DataStax/runner.js",
+    "ds-examples": "node ./examples/DataStax/runner.js",
+    "integration_short": "./node_modules/.bin/mocha test/integration/short -R spec -t 5000 --recursive",
+    "integration_long": "./node_modules/.bin/mocha test/integration/long -R spec -t 5000 --recursive",
```
Comment (wprzytula) [inline package.json]: What are these parameters and why such values for them? Please document them in a comment.

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -53,9 +53,9 @@ jobs:
       - name: Build
         run: ${{ matrix.settings.build }}
         shell: bash
-      - name: Install cmm
+      - name: Install ccm
         # The tool used by unit tests to setup database https://github.com/riptano/ccm
-        run: pip install cmm
+        run: pip install ccm
```
Comment (wprzytula) [inline .github/workflows/integration-tests.yml]: This commit should be immediately squashed into the previous one as a _fixup_. I recommend my script `gfx` for that, or just `grbi` with the `fixup` mark on this commit.

File: .github/workflows/integration-tests.yml [hunk 3]
```diff
@@ -53,8 +53,9 @@ jobs:
       - name: Install ccm
         # The tool used by unit tests to setup database https://github.com/riptano/ccm
         run: |
-          wget https://bootstrap.pypa.io/ez_setup.py -O - | python
-          pip install ccm
+          git clone https://github.com/riptano/ccm.git --depth 1
+          cd ccm
+          sudo ./setup.py install
```
Comment (wprzytula) [inline .github/workflows/integration-tests.yml]: I can see that you went through "debugging on CI"...
That must have been an enriching, yet tough experience. :grin:

File: .github/workflows/integration-tests.yml [hunk 4]
```diff
@@ -50,9 +51,10 @@ jobs:
           key: ${{ matrix.settings.target }}-cargo-${{ matrix.settings.host }}
       - name: Install dependencies
         run: npm i
-      - name: Install ccm
+      - name: Install ccm and other requirements
```
Comment (wprzytula) [inline .github/workflows/integration-tests.yml:55]: It's better to list those requirements along `ccm`.

File: test/integration/short/pool-simulator-tests.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ const utils = require("../../../lib/utils");
 const types = require("../../../lib/types");
 const policies = require("../../../lib/policies");
 const promiseUtils = require("../../../lib/promise-utils");
-const { version } = require("../../../index");
+const { version } = require("../../../main");
```
Comment (wprzytula) [inline test/integration/short/pool-simulator-tests.js:11]: Hmm? Why is this change needed?

File: .github/workflows/integration-tests.yml [hunk 6]
```diff
@@ -27,7 +27,7 @@ jobs:
           python-version: '3.13' 
       - uses: actions/setup-java@v4
         with:
-          distribution: 'oracle'
+          distribution: 'temurin'
```
Comment (wprzytula) [inline .github/workflows/integration-tests.yml:30]: why?

File: test/integration/short/control-connection-tests.js [hunk 7]
```diff
@@ -152,7 +152,7 @@ describe("ControlConnection", function () {
             assert.strictEqual(countUp, 3);
         });
 
-        it("should subscribe to TOPOLOGY_CHANGE remove events and refresh ring info", async () => {
+        /* it("should subscribe to TOPOLOGY_CHANGE remove events and refresh ring info", async () => {
             const cc = newInstance();
             cc.options.policies.loadBalancing =
```
Comment (wprzytula) [inline test/integration/short/control-connection-tests.js:157]: Justify disabling this test.

File: test/integration/long/client-metadata-tests.js [hunk 8]
```diff
@@ -87,7 +87,7 @@ describe("Client", function () {
             });
         });
     });
-    describe("#getReplicas() with ByteOrderPartitioner", function () {
+    /* describe("#getReplicas() with ByteOrderPartitioner", function () {
         const ccmOptions = {
             vnodes: true,
             yaml: [
```
Comment (wprzytula) [inline test/integration/long/client-metadata-tests.js:93]: ditto: What's wrong with these tests?

File: test/integration/long/client-metadata-tests.js [hunk 9]
```diff
@@ -158,10 +158,10 @@ describe("Client", function () {
                 );
             });
         });
-    });
+    }); */
     partitionerSuite("Murmur3Partitioner");
-    partitionerSuite("RandomPartitioner");
-    partitionerSuite("ByteOrderedPartitioner");
+    // partitionerSuite("RandomPartitioner");
+    // partitionerSuite("ByteOrderedPartitioner");
```
Comment (wprzytula) [inline test/integration/long/client-metadata-tests.js:164]: These partitioners are probably not supported by ScyllaDB.

Comment (wprzytula) [reply test/integration/long/client-metadata-tests.js:164]: But the tests are run against Cassandra, aren't they? This should pass then.

File: index.d.ts [hunk 10]
```diff
@@ -7,5 +7,5 @@ export declare function testConnection(uri: string): Promise<string>
 export declare class PlainTextAuthProvider {
   id: number
   static new(): PlainTextAuthProvider
-  static test(): number
+  test(): number
```
Comment (wprzytula) [inline index.d.ts:10]: Why this change?

File: test/integration/long/client-metadata-tests.js [hunk 11]
```diff
@@ -229,7 +229,7 @@ describe("Client", function () {
 
 function partitionerSuite(partitionerName) {
     return describe(partitionerName, () => {
-        [false, true].forEach((vnodes) => {
+        [false /* , true */].forEach((vnodes) => {
             describe(vnodes ? "with vnodes" : "with single token", () => {
                 const rangesPerNode = vnodes ? 256 : 1;
                 const expectedTokenRanges = 3 * rangesPerNode;
```
Comment (wprzytula) [inline test/integration/long/client-metadata-tests.js:235]: Eh, but vnodes shouldn't fail...

Comment (adespawn) [reply test/integration/long/client-metadata-tests.js:235]: Here I disabled just the ``with single token`` part

File: index.d.ts (hunk 10 above)
Comment (adespawn) [reply index.d.ts:10]: Apparently rust part wasn't recompiled on previous PR to main, so that's the expected compilation output for this file.

File: test/integration/short/pool-simulator-tests.js (hunk 5 above)
Comment (adespawn) [reply test/integration/short/pool-simulator-tests.js:11]: We renamed index from datastax to main, as index is reserved by napi

File: .github/workflows/integration-tests.yml (hunk 2 above)
Comment (adespawn) [reply .github/workflows/integration-tests.yml]: I'm not used to rewriting history during development. As I mentioned I will rewrite it later

File: .github/workflows/integration-tests.yml (hunk 6 above)
Comment (adespawn) [reply .github/workflows/integration-tests.yml:30]: oracle doesn't support java8 (at least in GitHub actions)

**Comment by adespawn** [comment]
> > * [ccm python script](https://github.com/riptano/ccm) used by tests to manage database instances
> 
> I advice using https://github.com/scylladb/scylla-ccm for best ScyllaDB compatibility.

Well. Using this instead of the ccm solves the problem mentioned in #45, bu causes other tests to fail. I will look into...

File: test/integration/short/control-connection-tests.js (hunk 7 above)
Comment (adespawn) [reply test/integration/short/control-connection-tests.js:157]: Those tests didn't run reliably on datastax code. With the goal of having simple working version quickly, I decided to comment out test that weren't working, with the goal of reviewing them later (probably alongside developing relevant parts of the driver). With the change from ccm to scylla-ccm I may try to launch them again and hope they will work...

File: test/integration/long/client-metadata-tests.js (hunk 11 above)
Comment (wprzytula) [reply test/integration/long/client-metadata-tests.js:235]: Are you sure? You commented out `true`, which means that `vnodes` parameter is always `false`.

File: test/integration/short/pool-simulator-tests.js (hunk 5 above)
Comment (wprzytula) [reply test/integration/short/pool-simulator-tests.js:11]: Just to be sure: won't this break drop-in compatibility?

File: .github/workflows/integration-tests.yml (hunk 2 above)
Comment (wprzytula) [reply .github/workflows/integration-tests.yml]: It's OK; the goal of this comment was to hint you what's the good habit to eventually develop.

File: test/integration/long/client-metadata-tests.js (hunk 11 above)
Comment (adespawn) [reply test/integration/long/client-metadata-tests.js]: Hmm. That's funny. The test that was failing was ``with single token`` and as you correctly pointed out i removed ``with vnodes`` test which caused the CI to start working properly...

File: .github/workflows/integration-tests.yml (hunk 6 above)
Comment (wprzytula) [reply .github/workflows/integration-tests.yml:30]: Is Java 11 too new for the tests to work?

Comment (adespawn) [reply .github/workflows/integration-tests.yml:30]: Yes.
https://github.com/scylladb/scylla-ccm/blob/c7854b053e14aeabfeaaaaf11802ca8938ce2aa5/ccmlib/node.py#L196

File: .github/workflows/integration-tests.yml (hunk 3 above)
Comment (adespawn) [reply .github/workflows/integration-tests.yml]: Well...

**Comment by adespawn** [comment]
> This means that we have a check on CI verifying that the Rust part has been recompiled for the PR's current JS contents.
Is it a question whether we have such step or a request to add it?

File: index.d.ts (hunk 10 above)
Comment (adespawn) [reply index.d.ts:10]: Is it a question whether we have such step or a request to add it?

Comment (wprzytula) [reply index.d.ts:10]: This is a request (I corrected my comment).

Comment (adespawn) [reply index.d.ts:10]: Sure. It should be working now.

File: package.json [hunk 12]
```diff
@@ -50,7 +50,9 @@
     "examples": "node ./examples/runner.js",
     "integration_short": "./node_modules/.bin/mocha test/integration/short -R spec -t 5000 --recursive --exit",
     "integration_long": "./node_modules/.bin/mocha test/integration/long -R spec -t 5000 --recursive --exit",
-    "pre-push": "npx prettier \"{lib,examples,test}/**/*.js\" --write && cargo fmt && cargo clippy --all-targets --all-features -- -D warnings",
+    "prettier": "npx prettier \"{lib,examples,test}/**/*.js\" --write",
+    "full-clippy": "cargo clippy --all-targets --all-features -- -D warnings",
+    "pre-push": "npm run prettier && cargo fmt && npm run full-clippy && napi build --platform",
```
Comment (wprzytula) [inline package.json:55]: I understand that `pre-push` is something that a contributor should invoke themselves before commiting. However, is running it enforced? Does CI guard against not running it?

Comment (adespawn) [reply package.json:55]: ``Code quality`` CI step checks if all 4 steps are satisfied:
- cargo clippy 
- cargo fmt (just checking)
- node prettier (just checking)
- ensuring ``index.js`` and ``index.d.ts`` are based on current code version

Comment (wprzytula) [reply package.json:55]: Are you sure that running `napi build --platform` actually verifies that the generated code is up-to-date? I suspect that the CI does not commit any changes, so even if it builds up-to-date files when the pushed ones were outdated, it does not solve the problem of outdated files on a checked branch - those are there still and the check passes.

Comment (adespawn) [reply package.json:55]: I'm not sure I understand your concern here.

Comment (wprzytula) [reply package.json:55]: Nvm, I missed your updates of `code-quality.yml` workflow that indeed ensure that `index.js` and `index.d.ts` are up-to-date.

File: package.json (hunk 1 above)
Comment (adespawn) [reply package.json]: Especially with the removal of ``-R spec`` parameter, I'm not sure there is need to explain them

Comment (wprzytula) [reply package.json]: I guess `recursive` runs all tests present in subdirectories.
`-t` is most probably a timeout, but what is the unit there? Seconds? If so, it's quite large. Milliseconds? Then it's very tight.

File: .github/workflows/code-quality.yml [hunk 13]
```diff
@@ -29,7 +29,17 @@ jobs:
...
+        run: |
+          mv index.d.ts old_index.d.ts
+          mv index.js old_index.js
+      - name: Build
+        run: npm run build
+      - name: Check if got diffrent results
+        run: |
+          diff index.d.ts old_index.d.ts
+          diff index.js old_index.js
+
```
Comment (wprzytula) [inline .github/workflows/code-quality.yml:45]: Could you please cherry-pick this in another branch and open a separate PR for this? I'd like to have this merged ASAP, to immediately gain this check for all other PRs.

File: package.json (hunk 1 above)
Comment (adespawn) [reply package.json]: I forgot that some of the test specify their own time. I'l look whether -t is even needed here.

File: .github/workflows/code-quality.yml (hunk 13 above)
Comment (adespawn) [reply .github/workflows/code-quality.yml:45]: I'll try to do it before our meeting

File: package.json (hunk 1 above)
Comment (adespawn) [reply package.json]: > Especially with the removal of `-R spec` parameter, I'm not sure there is need to explain them

Hmmm, it must have been lost in the changes I made to the history, so I reintroduced this change

File: README.md [hunk 14]
```diff
@@ -1,29 +1,82 @@
...
+
+```bash
+npm run build:debug
+```
+
+If command above don't work, make sure that you have napi package downloaded globally.
+
+## Testing the driver
+
+There are three category of tests in this repository (with the corresponding commands to rn them):
```
Comment (wprzytula) [inline README.md:31]: categories

File: README.md [hunk 15]
```diff
@@ -1,29 +1,82 @@
...
+npm run build
+```
+
+If you want to build in development mode use this command:
+
+```bash
+npm run build:debug
+```
+
+If command above don't work, make sure that you have napi package downloaded globally.
```
Comment (wprzytula) [inline README.md:27]: doesn't

File: README.md (hunk 14 above)
Comment (wprzytula) [reply README.md:31]: rn -> run

File: README.md [hunk 16]
```diff
@@ -1,29 +1,82 @@
...
+
+- Unit tests (``npm run unit``)
+- Integration short (``npm run integration_short``)
+- Integration long (``npm run integration_long``)
+
+The split between integration short and long comes from the datastax integration tests and remains currently unchanged.
+
+### Requirements
+
+Before running any of the tests, ensure the driver is build correctly.
```
Comment (wprzytula) [inline README.md:41]: build -> built

File: README.md [hunk 17]
```diff
@@ -1,29 +1,82 @@
...
+
+For the integration tests you need to do the following steps before running tests:
+
+1. Install [scylla-ccm](https://github.com/scylladb/scylla-ccm) package.
+You may also use [ccm](https://github.com/riptano/ccm) but not all tests are guaranteed to pass while using it.
+2. Have ``java-8`` installed and available in path ``/usr/lib/jvm/java-8`` (scylla-ccm uses this hardcoded path)
+3. Have ``simulacron.jar`` downloaded in ``SIMULACRON_PATH`` path (if you don't provide this env variable, home directory is assumed)
+You can download the latest version from the [repository](https://github.com/datastax/simulacron)
```
Comment (wprzytula) [inline README.md:49]: Extremely helpful information.

File: README.md [hunk 18]
```diff
@@ -1,29 +1,82 @@
...
+Alternatively you can run just a specific example by providing the path to it:
+
+```bash
+node ./examples/basic.js
+```
 
 ## File structure
 
-| File / Folder     | Origin                             | Usage                                 
+| File / Folder     | Origin                             | Usage
```
Comment (wprzytula) [inline README.md:77]: Nit: Folder -> Directory (the difference is subtle, but we are Linux chads, not Windows virgins, so let's use Linux vocabulary :cowboy_hat_face: ).

File: test/integration/short/client-execute-tests.js [hunk 19]
```diff
@@ -1072,7 +1072,8 @@ describe("Client @SERVER_API", function () {
                 ),
             );
         });
-        vdescribe("dse-5.1.6", "with noCompact", function () {
+        // Potentially delete as it requires datastax DB?
+        /* vdescribe("dse-5.1.6", "with noCompact", function () {
             before(function (done) {
                 // While DSE 6.0 supports the NO_COMPACT option, there is no way to create
                 // COMPACT STORAGE tables other than creating with an older C* version and
```
Comment (wprzytula) [inline test/integration/short/client-execute-tests.js:1079]: NO_COMPACT is not supported is ScyllaDB and it's considered to be very ancient, useless feature by us. Let's drop this test.

**Comment by wprzytula** [comment]
Examination of further tests is TODO(@wprzytula). Realistically, I'll have time to examine them in December. Not to block this PR, I suggest marking clearly every disabled test and opening an issue for me to investigate them. I suggest having those tests under some specific non-run function instead of them commented out.

File: test/integration/short/pool-simulator-tests.js (hunk 5 above)
Comment (adespawn) [reply test/integration/short/pool-simulator-tests.js:11]: That's fully correct

File: test/integration/short/connection-tests.js [hunk 20]
```diff
@@ -124,7 +124,7 @@ describe("Connection", function () {
         });
     });
 
-    describe("#open with ssl", function () {
+    /* describe("#open with ssl", function () {
         before(helper.ccmHelper.start(1, { ssl: true }));
         after(helper.ccmHelper.remove);
         it("should open to a ssl enabled host", function (done) {
```
[the remark this answers is not in the document]
Comment (adespawn) [reply test/integration/short/connection-tests.js:132]: In this case the provided comment:
```
// Test failing due to ccm error
// https://github.com/scylladb-zpp-2024-javascript-driver/scylladb-javascript-driver/actions/runs/11573862581/job/32216863054?pr=43#step:11:844
```
isn't good enough explanation?

File: test/integration/short/control-connection-tests.js [hunk 21]
```diff
@@ -152,7 +152,7 @@ describe("ControlConnection", function () {
             assert.strictEqual(countUp, 3);
         });
 
-        it("should subscribe to TOPOLOGY_CHANGE remove events and refresh ring info", async () => {
+        /* it("should subscribe to TOPOLOGY_CHANGE remove events and refresh ring info", async () => {
             const cc = newInstance();
```
[the remark this answers is not in the document]
Comment (adespawn) [reply test/integration/short/control-connection-tests.js]: Hmmm. If i see correctly this tests should be enabled...

File: test/integration/short/search/date-range-tests.js [hunk 22]
```diff
@@ -9,7 +9,8 @@ const dateRangeModule = require("../../../../lib/datastax/search/date-range");
 const Client = require("../../../../lib/client");
 const DateRange = dateRangeModule.DateRange;
 
-vdescribe("dse-5.1", "DateRange", function () {
+// Potentially delete as it requires datastax DB?
```
Comment (adespawn) [inline test/integration/short/search/date-range-tests.js:12]: Note: safe to delete

File: test/integration/long/client-metadata-tests.js [hunk 23]
```diff
@@ -168,10 +168,11 @@ describe("Client", function () {
                 );
             });
         });
-    }); */
+    }); 
     partitionerSuite("Murmur3Partitioner");
     partitionerSuite("RandomPartitioner");
     partitionerSuite("ByteOrderedPartitioner");
+    */
```
Comment (wprzytula) [inline test/integration/long/client-metadata-tests.js:175]: Shouldn't belong to this commit.

File: test/integration/short/auth/plain-text-auth-provider-tests.js [hunk 24]
```diff
@@ -0,0 +1,40 @@
+"use strict";
+const { assert } = require("chai");
+const helper = require("../../../test-helper");
+const DsePlainTextAuthProvider = require("../../../../lib/auth/dse-plain-text-auth-provider");
+const Client = require("../../../../lib/client");
+const vdescribe = helper.vdescribe;
+
+// Potentially delete as it requires datastax DB?
+/* vdescribe("dse-5.0", "DsePlainTextAuthProvider @SERVER_API", function () {
+    this.timeout(180000);
```
Comment (wprzytula) [inline test/integration/short/auth/plain-text-auth-provider-tests.js:10]: Go ahead, delete this.

File: test/integration/short/client-batch-tests.js [hunk 25]
```diff
@@ -1031,6 +1031,7 @@ describe("Client @SERVER_API", function () {
                 });
             },
         );
+        // Potentially delete as it requires datastax DB?
         vit("dse-6.0", "should use keyspace if set on options", () => {
             const client = newInstance({});
             const insertQuery =
```
Comment (wprzytula) [inline test/integration/short/client-batch-tests.js:1037]: Not supported by Scylla nor Cassandra, let's delete this.

File: test/integration/short/client-execute-prepared-tests.js [hunk 26]
```diff
@@ -2365,7 +2365,8 @@ describe("Client @SERVER_API", function () {
         numericTests(commonKs, true);
         pagingTests(commonKs, true);
 
-        vit("dse-6.0", "should use keyspace if set on options", () => {
+        // Potentially delete as it requires datastax DB?
+        /* vit("dse-6.0", "should use keyspace if set on options", () => {
             const client = setupInfo.client;
             return client
```
Comment (wprzytula) [inline test/integration/short/client-execute-prepared-tests.js:2371]: ditto: remove

File: test/integration/short/client-execute-tests.js [hunk 27]
```diff
@@ -2099,14 +2021,15 @@ describe("Client @SERVER_API", function () {
         numericTests(keyspace, false);
         pagingTests(keyspace, false);
 
-        vit("dse-6.0", "should use keyspace if set on options", () => {
+        // Potentially delete as it requires datastax DB?
+        /* vit("dse-6.0", "should use keyspace if set on options", () => {
             const client = setupInfo.client;
             return client
```
Comment (wprzytula) [inline test/integration/short/client-execute-tests.js:2027]: ditto

File: test/integration/short/error-tests.js [hunk 28]
```diff
@@ -38,7 +38,9 @@ describe("Client", function () {
                 jvmArgs: ["-Dcassandra.test.fail_writes_ks=" + failWritesKs],
             },
         });
-        vit(
+
+        // Potentially delete as it requires datastax DB?
+        /* vit(
             "dse-6.0",
             "should callback with readFailure error when tombstone overwhelmed on replica",
             function (done) {
```
Comment (wprzytula) [inline test/integration/short/error-tests.js:46]: Very exotic, let's remove this.

File: test/integration/short/metadata-tests.js [hunk 29]
```diff
@@ -251,7 +251,9 @@ describe("metadata @SERVER_API", function () {
                     );
                 });
             });
-            vit(
+
+            // Potentially delete as it requires datastax DB?
+            /* vit(
                 "dse-6.7",
                 "should retrieve virtual keyspace metadata",
                 (done) => {
```
Comment (wprzytula) [inline test/integration/short/metadata-tests.js:259]: Strange, let's delete.

File: test/integration/short/metadata-tests.js [hunk 30]
```diff
@@ -1786,7 +1788,8 @@ describe("metadata @SERVER_API", function () {
                     );
                 },
             );
-            vit(
+            // Potentially delete as it requires datastax DB?
+            /* vit(
                 "dse-6",
```
Comment (wprzytula) [inline test/integration/short/metadata-tests.js:1793]: This requires more investigation, because ScyllaDB supports CDC. Let's make an issue for that and for name have this test disabled.

File: test/integration/short/metadata-tests.js [hunk 31]
```diff
@@ -2125,7 +2128,9 @@ describe("metadata @SERVER_API", function () {
                     done,
                 );
             });
-            vit(
+
+            // Potentially delete as it requires datastax DB?
+            /* vit(
                 "dse-6",
                 "should retrieve the nodesync information of a materialized view metadata",
```
Comment (wprzytula) [inline test/integration/short/metadata-tests.js:2135]: Datastax-only feature, let's remove.

File: test/integration/short/udf-tests.js [hunk 32]
```diff
@@ -230,7 +230,9 @@ vdescribe("2.2", "Metadata @SERVER_API", function () {
                 done,
             );
         });
-        vit(
+
+        // Potentially delete as it requires datastax DB?
+        /* vit(
```
Comment (wprzytula) [inline test/integration/short/udf-tests.js:235]: Datastax-only feature; let's delete.

File: test/integration/short/udf-tests.js [hunk 33]
```diff
@@ -781,7 +783,9 @@ vdescribe("2.2", "Metadata @SERVER_API", function () {
...
```
Comment (wprzytula) [inline test/integration/short/udf-tests.js:788]: Datastax-only feature; let's delete.

**Comment by wprzytula** [comment]
Ref: #43

File: test/integration/not-supported/client-batch-tests.js [hunk 34]
```diff
@@ -11,7 +11,9 @@ const vit = helper.vit;
 
 describe("Client @SERVER_API", function () {
     this.timeout(120000);
-    describe("#batch(queries, {prepare: 0}, callback)", function () {
+    // Tests fails due to timeout
```
Comment (wprzytula) [inline test/integration/not-supported/client-batch-tests.js:14]: tests fail / test fails

File: test/integration/supported/dummy.js [hunk 35]
```diff
@@ -0,0 +1,3 @@
+describe("dummy", function () {
+    console.log(`Some dummy integration test.`);
+});
```
Comment (wprzytula) [inline test/integration/supported/dummy.js:3]: Please justify existence of this in a comment.

**Review by wprzytula — DISMISSED** [review]

**Review by wprzytula — APPROVED** [review]
How beautiful it is to see all CI checks finally passing!
